### PR TITLE
Transient Forceloads

### PR DESCRIPTION
**Pull request by raymoo**
Adds a flag to forceload_block which lets you turn off persistence for that forceload.

### CODE REVIEW

File: builtin/game/forceloading.lua [hunk 1]
```diff
@@ -15,32 +16,54 @@ local function get_blockpos(pos)
...
-function core.forceload_free_block(pos)
+function core.forceload_free_block(pos, transient)
 	local blockpos = get_blockpos(pos)
 	local hash = core.hash_node_position(blockpos)
-	if blocks_forceloaded[hash] == nil then return end
-	if blocks_forceloaded[hash] > 1 then
-		blocks_forceloaded[hash] = blocks_forceloaded[hash] - 1
+	local relevant_table
+	local other_table
+	if transient then
```
Comment (est31) [inline builtin/game/forceloading.lua]: maybe you can export this into a little (local) helper function?

**Comment by est31** [comment]
Otherwise :+1:

**Comment by est31** [comment]
Related issue: #4470

File: builtin/game/forceloading.lua (hunk 1 above)
Comment (raymoo) [inline builtin/game/forceloading.lua]: Do you mean to make a function that takes a boolean and returns relevant_table, other_table?

**Comment by raymoo** [comment]
I addressed the line comment, I think.

**Comment by t4im** [comment]
Looks good to me. The conditions around block incrementing/decrementing took a moment to check, but it looks correct to me.

**Comment by sfan5** [comment]
:+1:

**Comment by paramat** [comment]
2516c516bc018e2ff58c7d07c5f2ee8fd5f67167

**Pull request closed without merging**